extensions/transferdetails/service.go

```diff
@@ -0,0 +1,284 @@
+	if err != nil {
+		return nil, err
+	}
+
+	model, err = s.GetCurrentVersion(ctx, docID)
+	if err != nil {
+		log.Error(err)
+		return nil, documents.ErrDocumentNotFound
+	}
+	attributes, err := extensions.CreateAttributesList(model, *req.Data, transfersFieldKey, transfersLabel)
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO: StringsToDIDS
+	c, err := deriveDIDs(req.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	err = model.AddAttributes(
+		documents.CollaboratorsAccess{
+			ReadWriteCollaborators: c,
+		},
+		true,
+		attributes...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := CreateValidator()
+	err = validator.Validate(nil, model)
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentInvalid, err)
+	}
+
+	return model, nil
+}
+
+// DeriveFromUpdatePayload derives an updated TransferDetail from an UpdateTransferDetailRequest
+func (s service) DeriveFromUpdatePayload(ctx context.Context, req UpdateTransferDetailRequest) (model documents.Model, err error) {
+	var docID []byte
+	if req.DocumentID == "" {
+		return nil, documents.ErrDocumentIdentifier
+	}
+
+	docID, err = hexutil.Decode(req.DocumentID)
+	if err != nil {
+		return nil, err
+	}
+
+	model, err = s.GetCurrentVersion(ctx, docID)
+	if err != nil {
+		log.Error(err)
+		return nil, documents.ErrDocumentNotFound
+	}
+
+	idx, err := extensions.FindAttributeSetIDX(model, req.TransferID, transfersLabel, transferIDLabel, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	//TODO: StringsToDIDS
+	c, err := deriveDIDs(req.Data)
+	if err != nil {
+		return nil, err
+	}
+
+	// overwriting is not enough because it is not required that
+	// the TransferDetail payload contains all TransferDetail attributes
+	model, err = extensions.DeleteAttributesSet(model, TransferDetailData{}, idx, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	attributes, err := extensions.FillAttributeList(*req.Data, idx, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+
+	err = model.AddAttributes(
+		documents.CollaboratorsAccess{
+			ReadWriteCollaborators: c,
+		},
+		true,
+		attributes...,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	validator := CreateValidator()
+	err = validator.Validate(nil, model)
+	if err != nil {
+		return nil, errors.NewTypedError(documents.ErrDocumentInvalid, err)
+	}
+
+	return model, nil
+}
+
+// TODO: move to generic function in attribute utils
+func (s service) findTransfer(model documents.Model, transferID string) (*TransferDetailData, error) {
+	idx, err := extensions.FindAttributeSetIDX(model, transferID, transfersLabel, transferIDLabel, transfersFieldKey)
+	if err != nil {
+		return nil, err
+	}
+	return s.deriveTransferData(model, idx)
+}
+
+// TODO: move to generic function in attribute utils
+func (s service) deriveTransferData(model documents.Model, idx string) (*TransferDetailData, error) {
+	data := new(TransferDetailData)
+
+	types := reflect.TypeOf(*data)
```

Comment:
can be moved to extension package starting from this line

Reply:
will do it when funding is migrated and we can make things more generic